### PR TITLE
Add reduced motion variant styles

### LINKED ISSUE #18: Respect reduced motion settings
While the minimalist homepage is quite pleasant, the movement on it can be jarring for some folks. For these cases there is the [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion) media query that can be used to develop alternative (usually much more minimal to no) animations.

Originally I had overcomplicated the potential solution (you can see it in the revisions of this issue). There is a simple fix though: wait until the user submits the query, just like other forms. At this point, move the search bar. With reduced motion enabled, pop the search bar up to the top without a transition. This will solve the movement and will also remove the possibility that the movement happens while you are typing, as in the video below.

https://user-images.githubusercontent.com/24264157/178134795-6124b694-efc6-4200-a439-eaf3c32050a5.mp4

Anyway, love the project! I will be following it closely. If you ever want to pop over to [Openverse](https://github.com/wordpress/openverse) we're also doing a non-profit search engine thing, though admittedly with a specific niche and goal in mind.

Also, if you're happy to accept a PR and any have suggestions for which solution you'd like to see implemented, I'd be glad to take a go at it :slightly_smiling_face:

### PR DESCRIPTION
**Pull request by sarayourfriend**
Fixes #18 

Removes the "real time" search behavior and animation when the client has indicated that they prefer reduced motion. In that case, the search waits for them to "submit", as most searches these days do (the behavior will be familiar).

Additionally, increases the debounce timeout for when real time search is enabled and keeps the "submit" event for these clients as well so that if someone beats the timeout, they still see a responsive search.

I also added instructions in the README with a suggestion for how to run the app locally, hopefully saving everyone some time in the future :slightly_smiling_face:

### CODE REVIEW

**Comment by ColinEspinas** [comment]
Thanks for the contribution, 
I find this change interesting as this would also kind of fix #20 by just waiting for the user to submit.
I'll come back to this once I have more feedback on the whole "wait for submit vs realtime updates" situation.

> To test this I was serving the app locally using python's http.server module. For this to work I had to move the assets and the config.js into the src directory. Is there a better way to run the app locally that doesn't require those changes?

I just use the liveserver extension on VSCode, it works by serving the root of the repository and just navigating to the `/src` route. I would agree that this is not a great setup. We are already looking into changing the technologies we use for the front-end and the extension. As this pure vanilla front-end is great for a prototype but not ideal for contributors to engage with.

I will probably talk about it more on element, join us if you've got some time to chat.

File: src/components/organisms/search-bar.js [hunk 1]
```diff
@@ -1,37 +1,40 @@
 import define from '../../utils/define.js';
-import debounce from '../../utils/debounce.js';
 import config from '../../../config.js';
 import { globalBus } from '../../utils/events.js';
 
 const template = () => /*html*/`
-  <div class="search-bar">
+  <form class="search-bar" method="GET" action="/">
```
Comment (ColinEspinas) [inline src/components/organisms/search-bar.js]: I don't think we need the `method` and `action` attributes here. Did you include them for any specific reason?

**Review by ColinEspinas — COMMENTED** [review]
Just reviewed the changes, I don't get why those attributes are needed.

**Comment by sarayourfriend** [comment]
> just navigating to the /src route

Oh dear, that's quite obvious now :facepalm: I was serving the `src` directory specifically instead of the root of the repo :sweat_smile: 

For the attributes: I think they are not necessary and wouldn't ever be used in a context where this page actually loads on the browser, given the rendering depends entirely on client side JS. If it was an SSR'd page then the form would make the input work even if JavaScript wasn't enabled. Given that's not the case I'll remove those attributes.

**Comment by sarayourfriend** [comment]
We discussed this briefly in [#mwmbl:matrix.org chat](https://matrix.to/#/!MDsQmCFnIphgrxQipD:matrix.org/$8PaKxBPIhYoJIAleNlnrM6uK5OJjbbADyWv9m-0lmkI?via=matrix.org&via=matrix.tech-bloc-sea.dev&via=wiredentrypoint.xyz) and decided that an alternative approach to try would be to only wait for submit when reduced motion is enabled but also increase the debounce timeout for "real time search" higher than 100ms. I'll work on implementing that later today.

**Comment by sarayourfriend** [comment]
@ColinEspinas changes are up, at your leisure :grin:

File: src/components/organisms/search-bar.js [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import define from '../../utils/define.js';
 import config from '../../../config.js';
 import { globalBus } from '../../utils/events.js';
+import debounce from '../../utils/debounce.js'
+
+const prefersReducedMotion = window.matchMedia('(prefers-reduced-motion)').matches;
```
Comment (ColinEspinas) [inline src/components/organisms/search-bar.js:6]: If we just use this once maybe just putting this condition into the if statement would be better. Do you think putting this outside will serve any purpose?

Comment (sarayourfriend) [reply src/components/organisms/search-bar.js:6]: No run-time purpose. I find it more readable but I have no strong preference.

Comment (ColinEspinas) [reply src/components/organisms/search-bar.js:6]: Fine by me, will merge.

**Pull request merged**